include/svd_quda.h

```diff
@@ -0,0 +1,608 @@
+#pragma once
+
+#include <float.h>
+#include <math_helper.cuh>
+
+#define DEVICEHOST __device__ __host__
+#define SVDPREC 1e-11
+#define LOG2 0.69314718055994530942
+#define INVALID_DOUBLE (-DBL_MAX)
+
+namespace quda {
+
+  // FIXME replace this with hypot
+  template <class Cmplx>
+  inline DEVICEHOST typename std::remove_reference<decltype(Cmplx::x)>::type cabs(const Cmplx & z)
```

Comment:
guess this was probably just moved from another file but DEVICEHOST doesn't look very QUDA 2.0-y. But we can ignore that.

Reply:
I would be happy to make the global change replacing the use of `__device__ __host__` with a macro we use globally to denote parallel regions.  We could also defer this to a subsequent PR.

Reply:
Let's keep that for later. Just b/c it will reset my viewed count on github.